### PR TITLE
add data source for aws_route53_resolver_firewall_domain_list

### LINKED ISSUE #25492: add data source for aws_route53_resolver_firewall_domain_list
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Add a data source for the aws_route53_resolver_firewall_domain_list resource.

### New or Affected Resource(s)

* aws_route53_resolver_firewall_domain_list

### Potential Terraform Configuration

```hcl
data "aws_route53_resolver_firewall_domain_list" "this {
  firewall_domain_list_id = "examplexxx"
}
```

### References

* https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_GetFirewallDomainList.html

### PR DESCRIPTION
**Pull request by silvaalbert**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* 
Closes #25492

Output from acceptance testing:

```
$ make testacc TESTS='TestAccRoute53ResolverFirewallDomainListDataSource_basic' PKG=route53resolver TESTARGS=-short
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/route53resolver/... -v -count 1 -parallel 20 -run='TestAccRoute53ResolverFirewallDomainListDataSource_basic' -short -timeout 180m
=== RUN   TestAccRoute53ResolverFirewallDomainListDataSource_basic
--- PASS: TestAccRoute53ResolverFirewallDomainListDataSource_basic (97.34s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/route53resolver	100.808s

...
```

### CODE REVIEW

File: internal/service/route53resolver/firewall_domain_list_data_source.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceResolverFirewallDomainList() *schema.Resource {
```
Comment (AdamTylerLynch) [inline internal/service/route53resolver/firewall_domain_list_data_source.go:12]: "Resolver" is inferred from the package name. Also, the datasource name should match the corresponding Resource. In this case ResourceFirewallDomainList.

```suggestion
func DataSourceFirewallDomainList() *schema.Resource {
```

File: internal/service/route53resolver/firewall_domain_list_data_source.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	if err != nil {
+		return fmt.Errorf("error getting Route53 Firewall Domain List: %w", err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("no  Route53 Firewall Domain List found matching criteria; try different search")
+	}
+
+	d.SetId(aws.StringValue(output.FirewallDomainList.Id))
```
Comment (AdamTylerLynch) [inline internal/service/route53resolver/firewall_domain_list_data_source.go:78]: There are multiple attributes here that are not present in `ResourceFirewallDomainList`. I would recommend an additional PR to add these attributes to `ResourceFirewallDomainList`.

File: internal/service/route53resolver/firewall_domain_list_data_source.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	output, err := conn.GetFirewallDomainList(input)
+
+	if err != nil {
+		return fmt.Errorf("error getting Route53 Firewall Domain List: %w", err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("no  Route53 Firewall Domain List found matching criteria; try different search")
```
Comment (AdamTylerLynch) [inline internal/service/route53resolver/firewall_domain_list_data_source.go:75]: Please remove double spacing.

File: internal/service/route53resolver/firewall_domain_list_data_source.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			},
+			"status_message": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceResolverFirewallDomainListRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (AdamTylerLynch) [inline internal/service/route53resolver/firewall_domain_list_data_source.go:61]: Rename to match datasource and associated resource:

```suggestion
func dataSourceFirewallDomainListRead(d *schema.ResourceData, meta interface{}) error {
```

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
A few changes requested.

I would like to get @ewbankkit's opinion:

The ResourceFirewallDomainList exposes a list of domains, I feel that should be exposed in the following ways in the data source.

1. Domains attribute for the selected FirewallDomainList
2. Ability to filter FirewallDomainList(s) by associated Domain (www.example.com).

Thoughts?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccRoute53ResolverFirewallDomainListDataSource_' PKG=route53resolver ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/route53resolver/... -v -count 1 -parallel 2  -run=TestAccRoute53ResolverFirewallDomainListDataSource_ -timeout 180m
=== RUN   TestAccRoute53ResolverFirewallDomainListDataSource_basic
--- PASS: TestAccRoute53ResolverFirewallDomainListDataSource_basic (90.40s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/route53resolver	96.375s
```

**Comment by ewbankkit** [comment]
@silvaalbert Thanks for the contribution :tada: :clap:.

**Pull request merged**